### PR TITLE
feat(anta): Added testcase to validate the STP disabled VLANs

### LINKED ISSUE #862: Add the test case to verify whether STP is enabled or disabled for the specific vlan

### PR DESCRIPTION
**Pull request by geetanjalimanegslab**
# Description
This test performs the following checks:
        1. Verifies that the STP is configured.
        2. Verifies that the specified VLAN is present on the device.
        3. Verifies that the STP is disabled for the specified VLANs.

    Expected Results
    ----------------
    * Success: The test will pass if STP is disabled for the specified VLANs.
    * Failure: The test will fail, if STP is enabled for the specified VLANs

Fixes  #862 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

**Review by vitthalmagadum — CHANGES_REQUESTED** [review]

File: anta/tests/stp.py [hunk 1]
```diff
@@ -309,3 +309,62 @@ def test(self) -> None:
...
+
+    This test performs the following checks:
+        1. Verifies that the STP is configured.
+        2. Verifies that the specified VLAN is present on the device.
+        3. Verifies that the STP is disabled for the specified VLANs.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if STP is disabled for the specified VLANs.
+    * Failure: The test will fail, if STP is enabled for the specified VLANs.
```
[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/stp.py:325]: * Success: The test will pass if all of the following conditions are met:
      - STP is properly configured on the device.
      - The specified VLAN(s) exist on the device.
      - STP is confirmed to be disabled for all the specified VLAN(s).

**Review by vitthalmagadum — APPROVED** [review]
Please update succes and failure in docstring.

**Review by vitthalmagadum — APPROVED** [review]
LGTM

File: anta/tests/stp.py [hunk 2]
```diff
@@ -309,3 +309,67 @@ def test(self) -> None:
...
             self.result.is_success()
+
+
+class VerifySTPDisabledVlans(AntaTest):
+    """Verifies the STP disabled VLAN(s).
+
+    This test performs the following checks:
+
+        1. Verifies that the STP is configured.
+        2. Verifies that the specified VLAN is exist on the device.
```
Comment (carl-baillargeon) [inline anta/tests/stp.py:320]:
```suggestion
        2. Verifies that the specified VLAN(s) exist on the device.
```

File: anta/tests/stp.py [hunk 3]
```diff
@@ -309,3 +309,67 @@ def test(self) -> None:
...
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all of the following conditions are met:
+        - STP is properly configured on the device.
+        - The specified VLAN(s) exist on the device.
+        - STP is confirmed to be disabled for all the specified VLAN(s).
+    * Failure: The test will fail if any of the following condition is met:
+        - STP is not configured on the device.
+        - The specified VLAN(s) not exist on the device.
```
Comment (carl-baillargeon) [inline anta/tests/stp.py:331]:
```suggestion
        - The specified VLAN(s) do not exist on the device.
```

File: anta/tests/stp.py [hunk 4]
```diff
@@ -309,3 +309,67 @@ def test(self) -> None:
...
+        - STP is confirmed to be enabled for any of the specified VLAN(s).
```
Comment (carl-baillargeon) [inline anta/tests/stp.py:332]:
```suggestion
        - STP is enabled for any of the specified VLAN(s).
```

File: anta/tests/stp.py [hunk 5]
```diff
@@ -309,3 +309,67 @@ def test(self) -> None:
...
+    """
+
+    categories: ClassVar[list[str]] = ["stp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show spanning-tree vlan detail", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySTPDisabledVlans test."""
+
+        vlans: list[Vlan]
+        """List of STP disabled VLAN(s)"""
```
Comment (carl-baillargeon) [inline anta/tests/stp.py:352]:
```suggestion
        """List of STP disabled VLAN(s)."""
```

File: tests/units/anta_tests/test_stp.py [hunk 6]
```diff
@@ -486,4 +494,48 @@
...
+                    "6": {"spanningTreeVlanInstance": {"protocol": "mstp", "bridge": {}}},
+                    "4094": {"spanningTreeVlanInstance": {"protocol": "mstp", "bridge": {}}},
+                }
+            }
+        ],
+        "inputs": {"vlans": ["16", "4093"]},
+        "expected": {"result": "failure", "messages": ["VLAN: 16 - Not configured", "VLAN: 4093 - Not configured"]},
+    },
+    {
+        "name": "failure-stp-disabled-vlans",
```
Comment (carl-baillargeon) [inline tests/units/anta_tests/test_stp.py:527]:
```suggestion
        "name": "failure-vlans-enabled",
```

File: tests/units/anta_tests/test_stp.py [hunk 7]
```diff
@@ -486,4 +494,48 @@
...
+    },
+    {
+        "name": "failure-stp-not-configured",
+        "test": VerifySTPDisabledVlans,
+        "eos_data": [{"spanningTreeVlanInstances": {}}],
+        "inputs": {"vlans": ["6", "4094"]},
+        "expected": {"result": "failure", "messages": ["STP is not configured"]},
+    },
+    {
+        "name": "failure-stp-disabled-vlans-not-found",
```
Comment (carl-baillargeon) [inline tests/units/anta_tests/test_stp.py:512]:
```suggestion
        "name": "failure-vlans-not-found",
```

File: tests/units/anta_tests/test_stp.py [hunk 8]
```diff
@@ -486,4 +494,48 @@
         "inputs": {"threshold": 10},
         "expected": {"result": "failure", "messages": ["STP is not configured."]},
     },
+    {
+        "name": "success-stp-disabled-vlans",
```
Comment (carl-baillargeon) [inline tests/units/anta_tests/test_stp.py:498]:
```suggestion
        "name": "success",
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM. I will push some minor changes to docstrings. Thank you

**Pull request merged**